core/common/src/AuthorizationClient.ts

```diff
@@ -20,4 +20,10 @@ import { AccessToken } from "@itwin/core-bentley";
 export interface AuthorizationClient {
   /** Obtain an [[AccessToken]] for the currently authorized user, or blank string if no token is available. */
   getAccessToken(): Promise<AccessToken>;
+
+  /** [[BeEvent]] which fires when the client's [[AccessToken]] changes. */
+  readonly onAccessTokenChanged?: BeEvent<(token: AccessToken) => void>;
+
+  /** An [[AccessToken]] exists which is not expired */
+  isAuthorized?: boolean;
```

Comment:
i feel like this can be derived from the accessToken itself

i like the event, and think its necessary, and shouldve always been apart of the interface

Reply:
I agree, I don't think this is necessary on the base interface.

Reply:
fair enough, alternatively you could export a standard utility function for this case...

```ts
export async function isAuthorized(ac: AuthorizationClient) {
  return await ac.getAccessToken() !== "";
}
```

but I agree this is unnecessary, and I don't think studio needs it, we can just recommend inlining of that code. The event however would still be useful.

Reply:
are we in agreement we can close this pr then?

Reply:
well no... I'm saying we still would like this PR to add the event, just not add `isAuthorized`

Reply:
Coming right up

Reply:
Created a new PR so as to not have 50 commits merged from master: https://github.com/iTwin/itwinjs-core/pull/6122